.evergreen/start-mongoproxy.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -eux pipefail
+
+GOVERSION="${GOVERESION:-1.24}"
+GOPATH="${GOPATH:-$HOME/go}"
+
+# Detect OS
+OS="$(uname -s)"
+
+# If GOROOT is not set, determine it based on the OS and user-provided
+# GOVERSION.
+if [[ -z "${GOROOT:-}" ]]; then
+  case "$OS" in
+  Linux | Darwin)
+    GOROOT="/opt/golang/go${GOVERSION}"
+    ;;
+  MINGW* | MSYS* | CYGWIN*)
+    GOROOT="C:\\golang\\go${GOVERSION}"
+    ;;
+  *)
+    echo "unsupported OS: $OS"
+    exit 1
+    ;;
+  esac
+fi
```

Comment:
Not sure if this is my particular Go environment being messed up (I installed Go following the instructions [here](https://go.dev/doc/install), macos arm64) but my go installation is in `/usr/local/go` and the script fails when I run it.

If I remove the GOROOT detection and set `GOROOT=/usr/bin/go`, this script works as expected for me

Reply:
@baileympearson I think most people will have the GOROOT installed on Darwin so I can make that a separate case that points to /user/local/go. It's common to globally export GOROOT, in which case this switch condition wouldn't execute. As you note, something like this should work:

```
GOROOT=/usr/local/go MONGODB_URI="mongodb://localhost:27017"  bash start-mongoproxy.sh
```